src/Tools/dotnet-monitor/ConfigurationJsonWriter.cs

```diff
@@ -0,0 +1,214 @@
+                        ProcessChildSection(provider, nameof(FileSystemEgressProviderOptions.DirectoryPath), includeChildSections: false);
+                        ProcessChildSection(provider, nameof(FileSystemEgressProviderOptions.IntermediateDirectoryPath), includeChildSections: false);
+                        ProcessChildSection(provider, nameof(FileSystemEgressProviderOptions.CopyBufferSize), includeChildSections: false);
+                    }
+                    else
+                    {
+                        //Emit other egress entries, with redaction
+                        foreach (IConfigurationSection providerProperty in egressProperties)
+                        {
+                            if (string.Equals(providerProperty.Key, nameof(EgressConfigureOptions.CommonEgressProviderOptions.Type), StringComparison.OrdinalIgnoreCase))
+                            {
+                                ProcessSection(providerProperty, includeChildSections: false, redact: false);
+                            }
+                            else
+                            {
+                                ProcessSection(providerProperty, includeChildSections: true, redact: true);
+                            }
+                        }
+                    }
+                    _writer.WriteEndObject();
+                }
+                _writer.WriteEndObject();
+            }
+        }
+
+        private IConfigurationSection ProcessChildSection(IConfiguration parentSection, string key, bool includeChildSections = true, bool redact = false)
+        {
+            IConfigurationSection section = parentSection.GetSection(key);
+            if (!section.Exists())
+            {
+                _writer.WritePropertyName(key);
+                _writer.WriteStringValue("NOT PRESENT");
+                return null;
+            }
+
+            ProcessSection(section, includeChildSections, redact);
+
+            return section;
+        }
+
+        private void ProcessSection(IConfigurationSection section, bool includeChildSections = true, bool redact = false)
+        {
+            _writer.WritePropertyName(section.Key);
+
+            IEnumerable<IConfigurationSection> children = section.GetChildren();
+
+            //If we do not traverse the child sections, the caller is responsible for creating the value
+            if (includeChildSections && children.Any())
+            {
+                _writer.WriteStartObject();
+                foreach (IConfigurationSection child in children)
+                {
+                    ProcessSection(child, includeChildSections, redact);
+                }
+                _writer.WriteEndObject();
+            }
+            else
+            {
+                if (!children.Any())
+                {
+                    if (redact)
+                    {
+                        _writer.WriteStringValue("REDACTED");
```

Comment:
nit: Maybe make the "REDACTED" and "NOT PRESENT" value more likely to not be misinterpreted as actual values or collide with possible usage of "REDACTED" and "NOT PRESENT". Maybe something like:
- "REDACTED" -> "\<redacted>" or "!redacted!" or ":redacted:"
- "NOT PRESENT" -> "\<null>" or "!null!" or ":null:"

Something with delimiters that are more likely to make these not actual values that someone might put into configuration, but avoid $ and # since they have special meaning as JSON references.

Reply:
I initially tried <REDACTED> but this gets escaped n json. I will try some other characters.

Reply:
Using :REDACTED: and :NOT PRESENT:. Note null handling is already built into the system and a null value is different than a missing section.